react-delegationdashboard/src/helpers/entireBalance.ts

```diff
@@ -17,18 +18,37 @@ export default function entireBalance({
   decimals,
 }: EntireBalanceType) {
   const bnBalance = new BigNumber(balance);
+  const bnMinDust = new BigNumber(minDust);
   const bnGasPrice = new BigNumber(gasPrice);
   const bnGasLimit = new BigNumber(gasLimit);
-  const entireBalance = bnBalance.minus(bnGasPrice.times(bnGasLimit));
-  if (entireBalance.comparedTo(0) === 1) {
-    const input = entireBalance.toString(10);
-    return denominate({
-      input,
-      denomination,
-      decimals,
-      showLastNonZeroDecimal: true,
-      addCommas: false,
-    });
-  }
-  return '0';
+  const entireBn = bnBalance.minus(bnGasPrice.times(bnGasLimit));
+  const entireBnMinusDust = entireBn.minus(bnMinDust);
+
+  const entireBalance =
+    // entireBalance >= 0
+    entireBn.comparedTo(0) === 1
+      ? denominate({
+          input: entireBn.toString(10),
+          denomination,
+          decimals,
+          showLastNonZeroDecimal: true,
+          addCommas: false,
+        })
+      : '0';
+
+  const entireBalanceMinusDust =
```

Comment:
as discussed, do we really need this entireBalanceMinusDust value?

Reply:
For delegation we need it. It's a prevention that the user will still have some amount to undelegate or do further actions